src/main/java/uk/gov/hmcts/cmc/claimstore/utils/CalculateMonthIncrement.java

```diff
@@ -0,0 +1,19 @@
+package uk.gov.hmcts.cmc.claimstore.utils;
+
+import java.time.LocalDate;
+
+public final class CalculateMonthIncrement {
+
+    private CalculateMonthIncrement(){
+        //Static calculate month increment class, no instances required
+    }
+
+    public static LocalDate calculateMonthlyIncrement(LocalDate startDate) {
+        if (startDate == null) {
+            return null;
+        }
+        LocalDate futureMonth = startDate.plusMonths(1);
+        return startDate.getDayOfMonth() != futureMonth.getDayOfMonth()
+            && futureMonth.getDayOfMonth() == futureMonth.lengthOfMonth() ? futureMonth.plusDays(1) : futureMonth;
```

Comment:
Please write test cases for this conditions especially the complex conditional statement in the return statement.

Reply:
Just a example. If the startDate is Jan 31st, then plus 1 month will take it to Feb 28th. This will add one day and push the date to March 1st. is that correct?

Reply:
thanks @ellie-harrison saw the test case. Good work.